homeassistant/components/hassio/websocket_api.py

```diff
@@ -137,3 +144,45 @@ async def websocket_supervisor_api(
         )
     else:
         connection.send_result(msg[WS_ID], result.get(ATTR_DATA, {}))
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(WS_TYPE): "hassio/update/addon",
+        vol.Required("addon"): str,
+        vol.Required("backup"): bool,
+    }
+)
+@websocket_api.async_response
+async def websocket_update_addon(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Websocket handler to update an addon."""
+    addon_name: str | None = None
+    addon_version: str | None = None
+    addons: list = (get_supervisor_info(hass) or {}).get("addons", [])
+    for addon in addons:
+        if addon[ATTR_SLUG] == msg["addon"]:
+            addon_name = addon[ATTR_NAME]
+            addon_version = addon[ATTR_VERSION]
+            break
+    await update_addon(hass, msg["addon"], msg["backup"], addon_name, addon_version)
```

Comment:
Adding a WS API for an entity outside the entity integration isn't common practice. Why don't the frontend just call the corresponding service action for the entity?

Reply:
The user can update an add-on from the hassio panel even if the `update` entity is disabled.